zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java

```diff
@@ -133,36 +142,44 @@ String processedIndexHtml(Resource indexHtml) {
   }
 
   @Bean @Lazy ArmeriaServerConfigurator uiServerConfigurator(
+    CompressionProperties compressionProperties,
     IndexSwitchingService indexSwitchingService) throws IOException {
     ServerCacheControl maxAgeYear =
       new ServerCacheControlBuilder().maxAgeSeconds(TimeUnit.DAYS.toSeconds(365)).build();
     Service<HttpRequest, HttpResponse> uiFileService =
       HttpFileServiceBuilder.forClassPath("zipkin-ui").cacheControl(maxAgeYear).build()
-        .orElse(HttpFileServiceBuilder.forClassPath("zipkin-lens").cacheControl(maxAgeYear).build());
+        .orElse(
+          HttpFileServiceBuilder.forClassPath("zipkin-lens").cacheControl(maxAgeYear).build());
 
     byte[] config = new ObjectMapper().writeValueAsBytes(ui);
-
-    return sb -> sb
-      .service("/zipkin/config.json", HttpFileBuilder.of(HttpData.of(config))
+    return sb -> {
+      sb.service("/zipkin/config.json", HttpFileBuilder.of(HttpData.of(config))
         .cacheControl(new ServerCacheControlBuilder().maxAgeSeconds(600).build())
         .contentType(MediaType.JSON_UTF_8)
         .build()
-        .asService())
-      .service("/zipkin/index.html", indexSwitchingService)
+        .asService());
+
+      sb.serviceUnder("/zipkin/", uiFileService);
 
       // TODO This approach requires maintenance when new UI routes are added. Change to the following:
       // If the path is a a file w/an extension, treat normally.
       // Otherwise instead of returning 404, forward to the index.
       // See https://github.com/twitter/finatra/blob/458c6b639c3afb4e29873d123125eeeb2b02e2cd/http/src/main/scala/com/twitter/finatra/http/response/ResponseBuilder.scala#L321
-      .service("/zipkin/", indexSwitchingService)
```

Comment:
don't worry about the red.. I was just re-organizing as we no longer have to fluent everything